src/tests/conformance/messages.rs

```diff
@@ -51,3 +51,32 @@ async fn handshake(node_addr: SocketAddr) -> io::Result<TcpStream> {
 
     Ok(peer_stream)
 }
+
+#[tokio::test]
+async fn unsolicitation_listener() {
```

Comment:
As this test isn't part of the index, we may want to ignore it by default (with `#[ignore]`)?